src/message_sender.c

```diff
@@ -188,6 +194,22 @@ static void log_message_chunk(MESSAGE_SENDER_INSTANCE* message_sender, const cha
 #endif
 }
 
+// Auxiliary function to verify if a given message is still in the pending messages queue.
+static bool is_message_in_queue(MESSAGE_SENDER_HANDLE message_sender, ASYNC_OPERATION_HANDLE message)
+{
+    bool result = false;
+
+    for (size_t i = 0; i < message_sender->message_count; i++)
+    {
+        if (message_sender->messages[i] == message)
+        {
+            result = true;
```

Comment:
```suggestion
            result = true;
            break;
```

Reply:
Gosh, absolutely correct. Thanks for catching that.